core/src/main/scala/com/microsoft/azure/synapse/ml/logging/Usage/UsageUtils.scala

```diff
@@ -0,0 +1,69 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.logging.Usage
+
+import com.microsoft.azure.synapse.ml.logging.common.SparkHadoopUtils.getHadoopConfig
+import com.microsoft.azure.synapse.ml.logging.common.WebUtils.usagePost
+import com.microsoft.azure.synapse.ml.logging.SynapseMLLogging
+import com.microsoft.azure.synapse.ml.logging.Usage.FabricConstants._
+import com.microsoft.azure.synapse.ml.logging.Usage.HostEndpointUtils._
+import com.microsoft.azure.synapse.ml.logging.Usage.TokenUtils.getAccessToken
+import java.time.Instant
+import java.util.UUID
+import org.apache.spark.sql.SparkSession
+import spray.json._
+import spray.json.DefaultJsonProtocol._
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+
+object UsageTelemetry {
+  private val SC = SparkSession.builder().getOrCreate().sparkContext
+  private val CapacityId = getHadoopConfig("trident.capacity.id", SC)
+  val WorkspaceId: String = getHadoopConfig("trident.artifact.workspace.id", SC)
+  private val PbiEnv = SC.getConf.get("spark.trident.pbienv", "").toLowerCase()
+
+  private val SharedHost = getMlflowSharedHost(PbiEnv)
+  val SharedEndpoint = f"{SharedHost}/metadata/workspaces/{WorkspaceId}/artifacts"
+  private val WlHost = getMlflowWorkloadHost(PbiEnv, CapacityId, WorkspaceId, SharedHost)
+
+  def reportUsage(payload: FeatureUsagePayload): Unit = {
```

Comment:
maybe make everything in this pr either private or private[ml] so people dont accidentally mess up your state

Reply:
Done. Sending in next iteration.